aspnetcore/signalr/java-client.md

```diff
@@ -91,12 +91,43 @@ HubConnection hubConnection = HubConnectionBuilder.create("YOUR HUB URL HERE")
     })).build();
 ```
 
+::: moniker range=">= aspnetcore-5.0"
+
+### Passing Class information in Java
+
+When calling the `on`, `invoke`, or `stream` methods of `HubConnection` in the Java client, users should pass a `Type` object rather than a `Class<?>` object to describe any generic `Object` passed to the method. A `Type` can be acquired using the provided `TypeReference` class. For example, using a custom generic class named `Foo<T>`, the following code gets the `Type`:
+
+```java
+Type fooType = new TypeReference<Foo<String>>() { }).getType();
+```
+
+For non-generics, such as primitives or other non-parameterized types like `String`, you can simply use the built-in `.class`.
+
+When calling one of these methods with one or more object types, use the generics syntax when invoking the method. For example, when registering an `on` handler for a method named `func`, which takes as arguments a String and a `Foo<String>` object, use the following code to set an action to print the arguments:
+
+```java
+hubConnection.<String, Foo<String>>on("func", (param1, param2) ->{
+    System.out.println(param1);
+    System.out.println(param2);
+}, String.class, fooType);
+```
+
+This convention is necessary because we can not retrieve complete information about complex types with the `Object.getClass` method due to type erasure in Java. For example, calling `getClass` on an `ArrayList<String>` would not return `Class<ArrayList<String>>`, but rather `Class<ArrayList>`, which does not give the deserializer enough information to correctly deserialize an incoming message. The same is true for custom objects.
+
+::: moniker-end
+
 ## Known limitations
 
-::: moniker range=">= aspnetcore-3.0"
+::: moniker range=">= aspnetcore-5.0"
+
+* Transport fallback and the Server Sent Events transport aren't supported.
+
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0" < aspnetcore-5.0"
```

Comment:
@Rick-Anderson the build is complaining about invalid characters here, did I do something wrong with the moniker sections?

Reply:
```suggestion
::: moniker range=">= aspnetcore-3.0 < aspnetcore-5.0"
```

Reply:
There's an extra double quote after "3.0". Removing that should fix the error.